packages/components/date-picker/src/date-picker-com/basic-date-table.vue

```diff
@@ -245,56 +247,38 @@ export default defineComponent({
       if (!date) return false
       return dayjs(date)
         .locale(lang.value)
-        .isSame(props.date.date(Number(cell.text)), 'day')
+        .isSame(props.date?.date(Number(cell.text)), 'day')
     }
 
     const getCellClasses = (cell) => {
-      const classes = []
-      if ((cell.type === 'normal' || cell.type === 'today') && !cell.disabled) {
+      const classes: string[] = []
+      const { selectionMode } = props
+      const { type, disabled, inRange, selected, customClass, start, end } =
+        cell
+      const isNormalOrToday = type === 'normal' || type === 'today'
+      if (isNormalOrToday && !disabled) {
         classes.push('available')
-        if (cell.type === 'today') {
-          classes.push('today')
-        }
+        type === 'today' && classes.push('today')
       } else {
-        classes.push(cell.type)
+        classes.push(type)
       }
 
       if (
-        props.selectionMode === 'day' &&
-        (cell.type === 'normal' || cell.type === 'today') &&
+        selectionMode === 'day' &&
+        isNormalOrToday &&
         cellMatchesDate(cell, props.parsedValue)
       ) {
         classes.push('current')
       }
 
-      if (
-        cell.inRange &&
-        (cell.type === 'normal' ||
-          cell.type === 'today' ||
-          props.selectionMode === 'week')
-      ) {
+      if (inRange && (isNormalOrToday || selectionMode === 'week')) {
         classes.push('in-range')
-
-        if (cell.start) {
-          classes.push('start-date')
-        }
-
-        if (cell.end) {
-          classes.push('end-date')
-        }
-      }
-
-      if (cell.disabled) {
-        classes.push('disabled')
-      }
-
-      if (cell.selected) {
-        classes.push('selected')
-      }
-
-      if (cell.customClass) {
-        classes.push(cell.customClass)
+        start && classes.push('start-date')
+        end && classes.push('end-date')
       }
+      disabled && classes.push('disabled')
+      selected && classes.push('selected')
+      customClass && classes.push(customClass)
```

Comment:
I'd prefer something like this:

```
const classes = {
  disabled,
  selected,
  customClass,
}

return Object.entries(classes).filter(([kls, val]) => val).map(([kls]) => kls).join(' ')
```

Reply:
This is well written, and other classes can also be handled in this way.